components/brave_new_tab_ui/components/default/braveToday/index.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+// Feature-specific components
+import * as BraveTodayElement from './default'
+
+// Feature Containers
+import CardIntro from './cards/cardIntro'
+
+const Content = React.lazy(() => import('./content'))
+
+type State = {
+  hasInteractionStarted: boolean
+}
+
+export type Props = {
+  feed?: BraveToday.Feed
+  publishers?: BraveToday.Publishers
+  articleToScrollTo?: BraveToday.FeedItem
+  displayedPageCount: number
+  onInteracting: (interacting: boolean) => any
+  onReadFeedItem: (item: BraveToday.FeedItem) => any
+  onAnotherPageNeeded: () => any
+  onCustomizeBraveToday: () => any
+}
+
+class BraveToday extends React.PureComponent<Props, State> {
+  braveTodayHitsViewportObserver: IntersectionObserver
+  scrollTriggerToFocusBraveToday: any // React.RefObject<any>
+
+  state = {
+    hasInteractionStarted: false
+  }
+
+  componentDidMount () {
+    const options = { root: null, rootMargin: '0px', threshold: 0.25 }
+
+    this.braveTodayHitsViewportObserver = new
+      IntersectionObserver(this.handleBraveTodayHitsViewportObserver, options)
```

Comment:
nice uses of intersection observers 😄 👍  I remember when those were added (Chromium 52? was before we hard forked Electron)

Reply:
This one is @cezaraugusto !